Content.Server/Alert/ServerAlertsSystem.cs

```diff
@@ -0,0 +1,66 @@
+using Content.Server.Gravity.EntitySystems;
+using Content.Shared.Alert;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.Log;
+
+namespace Content.Server.Alert;
+
+[UsedImplicitly]
+internal class ServerAlertsSystem : AlertsSystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AlertsComponent, ComponentStartup>(HandleComponentStartup);
+        SubscribeLocalEvent<AlertsComponent, ComponentRemove>(HandleComponentRemove);
+
+        SubscribeLocalEvent<AlertsComponent, ComponentGetState>(ClientAlertsGetState);
+        SubscribeNetworkEvent<ClickAlertEvent>(HandleClickAlert);
+    }
+
+    private static void ClientAlertsGetState(EntityUid uid, AlertsComponent component, ref ComponentGetState args)
+    {
+        args.State = new AlertsComponentState(component.Alerts);
+    }
+
+    private void HandleClickAlert(ClickAlertEvent msg, EntitySessionEventArgs args)
+    {
+        var player = args.SenderSession.AttachedEntity;
+        if (player is null || !EntityManager.TryGetComponent<AlertsComponent>(player, out var alertComp)) return;
+
+        if (!IsShowingAlert(player.Value, msg.Type))
+        {
+            Logger.DebugS("alert", "user {0} attempted to" +
+                                   " click alert {1} which is not currently showing for them",
+                EntityManager.GetComponent<MetaDataComponent>(player.Value).EntityName, msg.Type);
+            return;
+        }
+
+        if (!TryGet(msg.Type, out var alert))
+        {
+            Logger.WarningS("alert", "unrecognized encoded alert {0}", msg.Type);
+            return;
+        }
+
+        alert.OnClick?.AlertClicked(player.Value);
+    }
+
+    private static void HandleComponentStartup(EntityUid uid, AlertsComponent component, ComponentStartup args)
+    {
+        if (TryGet<WeightlessSystem>(out var weightlessSystem))
+            weightlessSystem.AddAlert(component);
+        else
+            Logger.WarningS("alert", "weightlesssystem not found");
+    }
```

Comment:
Instead of doing a TryGet for every component start up, why not just use the a system `[Dependency]`?

Reply:
The actual reason was because this file only exists because of the AlertsSystem dependency on the WeightlessSystem. I was going to switch the relationship and never got around to it, which would have removed all this code. If i can't switch it I will resolve this with the `[Dependency]`.